beacon/goclient/goclient.go

```diff
@@ -324,7 +325,7 @@ func (gc *GoClient) singleClientHooks() *eth2clienthttp.Hooks {
 				zap.String("version", nodeVersionResp.Data),
 			)
 
-			genesis, err := s.Genesis(ctx, &api.GenesisOpts{})
+			genesis, err := genesisImpl(ctx, gc.log, s)
```

Comment:
this could look better as `gc.genesis(ctx,s)` or `s.signlegenesis(ctx,s)`

Reply:
renamed to `genesisForClient`, let me know if it sounds bad to you